### DISCUSSION TITLE
Torrents won't show up after app restart

### DISCUSSION THREAD

**Discussion by dtitov** [anacrolix/torrent#492] [Q&A] [answered]
Hi. Might be an extremely silly question, but I'm a bit confused here. So, I'm trying to create a simple app based on this library. I add a torrent (via magnet), then I list all the torrents and find the added one there. But if I restart the application and list the torrents again, the list is... empty. Is it not supposed to save torrents in the embedded (I assume it's Bolt) storage? Or is this storage used only for the "runtime" and doesn't survive application restarts? Or maybe I should somehow manually store torrents if I want to find them after the restart? In my mind it's quite basic and essential functionality, so I just expected it to work out of the box, but perhaps I'm missing something here...

**Comment by anacrolix** [comment, marked as answer]
Hi @dtitov . You're correct, the only persistence provided by default is for the state of pieces for a given torrent and storage backend. If you add the same torrent later, and use the same piece completion and storage as in earlier runs, the assumed state of the data should be what it was previously (minus partially completed pieces, those aren't tracked).

You need to manually manage torrents, this isn't provided by the implementation.

**Reply by dtitov** [reply]
Oh, I see! So does it mean that upon adding a torrent I should additionally save some torrent metadata myself (elsewhere) and then upon restart I will add these torrents again to the client from the saved collection and the states will be restored (minus things you've mentioned)? Then the question would be: what exactly should be stored as torrent metadata for the subsequent restoration? And does it depend on the "source" (i.e. torrent file or magnet)? I assume it's `metainfo.MetaInfo`?

**Reply by dtitov** [reply]
Nvm, got it working with `metainfo.MetaInfo`. Thanks for quick help! :)
